### PR TITLE
Refactor windows code

### PR DESCRIPTION
**Pull request by ZehMatt**
This is basically a start for a long run of refactoring the windows code. This first PR moves a few functions out of the global scope into rct_struct for skeleton work. The idea is to first move out most of the functions into rct_struct and once all direct data access is eliminated and it has all the members functions we can create a new interface and move the entire code into the UI part of the project. Once that is completed the window manager should take care of all the window management also. The end goal is to have the window manager do all the heavy lifting while providing safe access to the window list at any time even during events.

Because this involves a lot of changes I decided to do this gradually.

### CODE REVIEW

File: src/openrct2/interface/Window_internal.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include "Window_internal.h"
+
+#include "../world/Sprite.h"
+
```
Comment (AaronVanGeffen) [inline src/openrct2/interface/Window_internal.cpp]: Can we move this file in its entirety to the `openrct2-ui` subproject? Most (if not all) of the callees are in the ui subproject as well, right?

Comment (ZehMatt) [reply src/openrct2/interface/Window_internal.cpp]: Thats planned for when everything is complete and have all the private members hidden behind an interface

**Comment by janisozaur** [comment]
Is this something you would want merged before next release? Are you going to keep extending this PR or create new ones?

**Comment by ZehMatt** [comment]
Not gonna extend this PR, its just a start for more consecutive PRs

**Pull request merged**